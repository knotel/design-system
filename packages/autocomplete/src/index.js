import React from 'react'
import PropTypes from 'prop-types'
import Downshift from 'downshift'
import styled from 'styled-components'
import {
  Flex,
  Card,
  Label as PclnLabel,
  Input as PclnInput
} from 'pcln-design-system'
import { themeGet } from 'styled-system'

export const AutocompleteContext = React.createContext()

export const withAutocomplete = (Component, mapProps) =>
  React.forwardRef((props, ref) => (
    <AutocompleteContext.Consumer
      children={state => (
        <Component ref={ref} {...props} {...mapProps({ ...props, ...state })} />
      )}
    />
  ))

export const Label = withAutocomplete(PclnLabel, ({ getLabelProps }) =>
  getLabelProps()
)
Label.isLabel = true

export const Input = withAutocomplete(PclnInput, ({ getInputProps }) =>
  getInputProps()
)
Input.isField = true

const MenuCard = styled(Card)`
  position: absolute;
  top: 100%;
  left: 0;
  right: 0;
  max-height: ${props => props.height};
  overflow-y: auto;
`

MenuCard.defaultProps = {
  bg: 'white',
  height: '256px',
  borderWidth: 0,
  boxShadowSize: 'lg',
  mt: 1
}

const MenuRoot = React.forwardRef((props, ref) => (
  <MenuCard {...props} innerRef={ref} />
))

export const Menu = ({ children, ...props }) => (
  <AutocompleteContext.Consumer
    children={({ match, isOpen, getMenuProps, inputValue }) => (
      <MenuRoot
        {...getMenuProps({
          ...props,
          isOpen
        })}
        children={
          isOpen
            ? React.Children.toArray(children)
                .filter(el => match(el.props.item, inputValue))
                .map((el, index) => React.cloneElement(el, { index }))
            : false
        }
      />
    )}
  />
)

const ItemRoot = styled(Flex)`
  cursor: pointer;
  &[aria-selected='true'] {
    background-color: ${themeGet('colors.lightBlue')};
  }
  &[data-highlighted] {
    color: ${themeGet('colors.white')};
    background-color: ${themeGet('colors.blue')};
    & svg {
      color: ${themeGet('colors.white')};
    }
  }
`

ItemRoot.defaultProps = {
  alignItems: 'center'
}

export const Item = props => (
  <AutocompleteContext.Consumer
    children={({ item, getItemProps, highlightedIndex }) => (
      <ItemRoot
        {...getItemProps({
          item,
          px: 2,
          py: 1,
          'data-highlighted':
            highlightedIndex === props.index ? true : undefined,
          ...props
        })}
      />
    )}
  />
)

export class Autocomplete extends React.Component {
  static Label = Label
  static Input = Input
  static Menu = Menu
  static Item = Item

  static propTypes = {
    match: PropTypes.func
  }

  static defaultProps = {
    match: () => true
  }

  render() {
    const { children, style, ...props } = this.props

    return (
      <Downshift
        {...props}
        children={state => (
          <div style={{ position: 'relative', ...style }}>
            <AutocompleteContext.Provider
              value={{ ...props, ...state }}
              children={children}
            />
          </div>
        )}
      />
    )
  }
}
