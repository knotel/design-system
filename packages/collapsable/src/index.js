// import React, { Component } from 'react'
import Collapsable from 'react-collapsible'

export default Collapsable
