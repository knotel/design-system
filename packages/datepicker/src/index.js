import styled from 'styled-components'
import {
  DateRangePicker,
  SingleDatePicker,
  DayPickerRangeController
} from 'react-dates'
import { space, color, themeGet } from 'styled-system'

export const Datepicker = styled(DateRangePicker)`
  ${space} ${color} 

  .CalendarDay__selected_span {
    background: #82e0aa; //background
    color: white; //text
    border: 1px solid ${themeGet('colors.alphablue')}; //default styles include a border
  }

  // Will edit selected date or the endpoints of a range of dates
  .CalendarDay__selected {
    background: ${themeGet('colors.alphablue')};
    color: white;
  }

  // Will edit when hovered over. _span style also has this property
  .CalendarDay__selected:hover {
    background: ${themeGet('colors.alphablue')};
    color: white;
  }

  .CalendarDay__hovered_span:hover,
  .CalendarDay__hovered_span {
    background: ${themeGet('colors.alphablue')};
  }
`

Datepicker.defaultProps = {
  color: 'blue'
}

Datepicker.propTypes = {
  ...space.propTypes,
  ...color.propTypes
}

export default Datepicker
