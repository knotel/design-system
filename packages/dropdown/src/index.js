import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { Box, Button, Card, Relative } from 'pcln-design-system'
import styled from 'styled-components'

class Dropdown extends Component {
  constructor(props) {
    super(props)

    this.state = {
      isOpen: props.isOpen
    }
  }

  toggleView = () => {
    const isOpen = !this.state.isOpen
    console.log('>', isOpen)
    this.setState({ isOpen })
  }

  render() {
    const { title, isOpen } = this.state
    return (
      <Box {...this.props}>
        <Button onClick={() => this.toggleView()}>{this.props.title}</Button>
        <Card p={2} width={240} hidden={!isOpen}>
          <OverflowContainer w={1} mb={3}>
            {this.props.children}
          </OverflowContainer>
        </Card>
      </Box>
    )
  }
}

const OverflowContainer = styled(Relative)`
  overflow: hidden;
`

Dropdown.propTypes = {
  children: PropTypes.arrayOf(PropTypes.node).isRequired,
  isOpen: PropTypes.bool
}

Dropdown.defaultProps = {
  children: [],
  isOpen: false
}

export default Dropdown
