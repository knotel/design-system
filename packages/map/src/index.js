import Map from 'react-map-gl'
export default Map
