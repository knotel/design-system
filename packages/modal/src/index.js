export { default as Modal } from './Modal'
export { default as SmallModalHeader } from './SmallModalHeader'
export { default as ModalHeader } from './ModalHeader'
export { default as ScrollLock } from './ScrollLock'
