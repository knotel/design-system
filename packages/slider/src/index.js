import React from 'react'
import styled from 'styled-components'
import Range from 'rc-slider/lib/Range'
import { space, color, themeGet } from 'styled-system'

export const Slider = styled(Range)`
  position: relative;
  height: 32px;
  padding-top: 12px;
  border-radius: 9999px;
  touch-action: none;

  ${space} ${color} & .rc-slider-rail, & .rc-slider-track {
    height: 8px;
  }
  & .rc-slider-handle {
    width: 32px;
    height: 32px;
    margin-left: -16px;
    margin-top: -12px;
  }

  & .rc-slider-rail {
    position: absolute;
    width: 100%;
    background-color: ${themeGet('colors.lightGray')};
    border-radius: 9999px;
  }

  & .rc-slider-track {
    position: absolute;
    left: 0;
    border-radius: 9999px;
    background-color: currentcolor;
  }

  & .rc-slider-handle {
    position: absolute;
    cursor: pointer;
    cursor: grab;
    border-radius: 9999px;
    border: solid 4px currentcolor;
    background-color: #fff;
    touch-action: pan-x;

    &:hover {
    }
    &:active {
    }
    &:focus {
      box-shadow: 0 0 0 2px ${themeGet('colors.alphablue')};
    }
  }

  &.rc-slider-disabled {
    color: ${themeGet('colors.borderGray')};
    .rc-slider-track {
    }
    .rc-slider-handle {
      box-shadow: none;
      cursor: default;
    }
  }
`

Slider.defaultProps = {
  allowCross: false,
  color: 'blue'
}

Slider.propTypes = {
  ...space.propTypes,
  ...color.propTypes
}

export default Slider
