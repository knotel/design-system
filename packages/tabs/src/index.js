import React from 'react'
import PropTypes from 'prop-types'
import { Flex, Text } from 'pcln-design-system'
import styled from 'styled-components'

const TabTitle = styled(Text)`
  border-bottom: ${props =>
    props.isActive ? `4px solid ${props.activeColor}` : 'none'};
  cursor: pointer;
`

class Tabs extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      activeTab: 0
    }
  }

  render() {
    const { activeColor, defaultColor, tabs } = this.props
    const children = React.Children.toArray(this.props.children)

    return (
      <React.Fragment>
        <Flex bg="lightGray" px={1} pt={2}>
          {tabs.map((tab, i) => (
            <TabTitle
              color={this.state.activeTab === i ? activeColor : defaultColor}
              key={i}
              mx={3}
              py={2}
              onClick={() => this.setState({ activeTab: i })}
              isActive={this.state.activeTab === i}
              {...this.props}
            >
              {tab.title}
            </TabTitle>
          ))}
        </Flex>
        <Flex bg="white" py={3} px={3}>
          {children[this.state.activeTab]}
        </Flex>
      </React.Fragment>
    )
  }
}

Tabs.defaultProps = {
  tabs: [],
  children: []
}

Tabs.propTypes = {
  activeColor: PropTypes.string,
  defaultColor: PropTypes.string,
  tabs: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired
    })
  ).isRequired,
  children: PropTypes.arrayOf(PropTypes.node).isRequired
}

export default Tabs
