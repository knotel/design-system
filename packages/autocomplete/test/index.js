import React from 'react'
import TestRenderer from 'react-test-renderer'
import us from 'us'
import { Autocomplete } from '../src'

describe('Autocomplete', () => {
  test('renders', () => {
    const json = TestRenderer.create(
      <Autocomplete>
        <Autocomplete.Label>Test</Autocomplete.Label>
        <Autocomplete.Input />
        <Autocomplete.Menu>
          {us.STATES.map(state => (
            <Autocomplete.Item
              key={state.name}
              item={state}
              children={state.name}
            />
          ))}
        </Autocomplete.Menu>
      </Autocomplete>
    ).toJSON()
    expect(json).toMatchSnapshot()
  })

  test('renders open', () => {
    const json = TestRenderer.create(
      <Autocomplete defaultIsOpen={true}>
        <Autocomplete.Menu>
          {us.STATES.map(state => (
            <Autocomplete.Item
              key={state.name}
              item={state}
              children={state.name}
            />
          ))}
        </Autocomplete.Menu>
      </Autocomplete>
    ).toJSON()
    const [menu] = json.children
    expect(menu.children.length > 0).toBe(true)
  })

  test('renders with highlightedIndex', () => {
    const json = TestRenderer.create(
      <Autocomplete defaultIsOpen={true} defaultHighlightedIndex={0}>
        <Autocomplete.Menu>
          {us.STATES.map(state => (
            <Autocomplete.Item
              key={state.name}
              item={state}
              children={state.name}
            />
          ))}
        </Autocomplete.Menu>
      </Autocomplete>
    ).toJSON()
    const [menu] = json.children
    const [firstItem] = menu.children
    expect(firstItem.props['data-highlighted']).toBe(true)
  })
})
