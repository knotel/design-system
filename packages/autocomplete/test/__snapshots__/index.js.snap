// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete renders 1`] = `
.c4 {
  margin-top: 4px;
  background-color: #fff;
}

.c3 {
  box-shadow: 0 0 2px 0 rgba(0,0,0,.08),0 4px 16px 0 rgba(0,0,0,.16);
  border: 0;
  border-radius: 2px;
}

.c1 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  display: block;
  width: 100%;
  font-family: inherit;
  font-size: 14px;
  color: inherit;
  background-color: transparent;
  border-radius: 2px;
  border-width: 0px;
  border-style: solid;
  border-color: #c0cad5;
  padding-top: 14px;
  padding-bottom: 14px;
  padding-left: 12px;
  padding-right: 12px;
  margin: 0;
  border-color: #c0cad5;
  box-shadow: 0 0 0 1px #c0cad5;
}

.c1::-webkit-input-placeholder {
  color: #4f6f8f;
}

.c1::-moz-placeholder {
  color: #4f6f8f;
}

.c1:-ms-input-placeholder {
  color: #4f6f8f;
}

.c1::placeholder {
  color: #4f6f8f;
}

.c1::-ms-clear {
  display: none;
}

.c1:focus {
  outline: 0;
  border-color: #007aff;
  box-shadow: 0 0 0 2px #007aff;
}

.c0 {
  font-size: 10px;
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
  display: block;
  width: 100%;
  margin: 0;
  font-size: 10px;
  color: #4f6f8f;
  font-weight: 700;
}

.c2 {
  position: absolute;
  top: 100%;
  left: 0;
  right: 0;
  max-height: 256px;
  overflow-y: auto;
}

<div
  aria-expanded={false}
  aria-haspopup="listbox"
  aria-labelledby="downshift-0-label"
  aria-owns={null}
  role="combobox"
  style={
    Object {
      "position": "relative",
    }
  }
>
  <label
    className="c0"
    color="gray"
    fontSize="10px"
    fontWeight="bold"
    htmlFor="downshift-0-input"
    id="downshift-0-label"
  >
    Test
  </label>
  <input
    aria-activedescendant={null}
    aria-autocomplete="list"
    aria-controls={null}
    aria-labelledby="downshift-0-label"
    autoComplete="off"
    className="c1"
    id="downshift-0-input"
    onBlur={[Function]}
    onChange={[Function]}
    onKeyDown={[Function]}
    value=""
  />
  <div
    aria-labelledby="downshift-0-label"
    className="c2 c3 c4"
    height="256px"
    id="downshift-0-menu"
    role="listbox"
  />
</div>
`;
